src/commands.js

```diff
@@ -173,7 +175,180 @@ module.exports = {
+      if (args.length === 0) {
+        res.info('usage: /baninfo KEY@SEQ')
+        return res.end()
+      }
+      cabal.core.banInfo(args[0], function (err, doc) {
+        if (err) return res.error(err)
+        res.info(Object.assign({}, doc, {
+          text: JSON.stringify(doc, 2, null)
+        }))
+        res.end()
+      })
+    }
+  },
+  banlist: {
+    help: () => 'list banned users per-channel or cabal-wide',
+    call: (cabal, res, arg) => {
+      var args = (arg || '').split(/\s+/)
+      var channel = args[0] || '@'
+      pump(cabal.core.moderation.listBans(channel), to.obj(write, end))
+      function write (row, enc, next) {
+        res.info(Object.assign({}, row, {
+          text: `banned ${row.type}: ${row.id}`
+        }))
+        next()
+      }
+      function end (next) {
+        res.end()
+        next()
+      }
+    }
+  },
+  role: {
+    help: () => 'direct access to get/set roles used by ban and moderation commands',
+    call: (cabal, res, arg) => {
+      if (!arg) {
+        res.info('usage: /role (get|set) (CHANNEL|@) KEY')
+        return res.end()
+      }
+      var args = arg.split(/\s+/)
+      var channel = args[1]
+      var key = args[2]
+      if (args[0] === 'get') {
+        cabal.core.moderation.getRole({ channel, key }, (err, role) => {
+          if (err) return res.error(err)
+          res.info({ role, text: role })
+          res.end()
+        })
+      } else {
+        res.error('not implemented')
+      }
+    }
+  },
+  mod: {
+    help: () => 'add, remove, or list moderators',
+    call: (cabal, res, arg) => {
+      function usage () {
+        res.info('usage: /mod (add|remove) (CHANNEL|@) KEY {REASON...}')
+        res.info('usage: /mod list (CHANNEL|@)')
+        res.end()
+      }
+      if (!arg) return usage()
+      var args = arg.split(/\s+/)
+      var channel = args[1] || '@'
+      var key = args[2]
+      var reason = args.slice(3).join(' ')
+      if (args[0] === 'add') {
+        cabal.core.addMod(key, { channel, reason }, (err) => {
+          if (err) res.error(err)
+          else res.end()
+        })
+      } else if (args[0] === 'remove') {
+        cabal.core.removeMod(key, { channel, reason }, (err) => {
+          if (err) res.error(err)
+          else res.end()
+        })
+      } else if (args[0] == 'list') {
+        pump(cabal.core.moderation.listMods(channel), to.obj(write, end))
+      } else {
+        usage()
+      }
+      function write (row, enc, next) {
+        res.info(Object.assign({}, row, {
+          text: `${row.id} [${row.role}]`
+        }))
+        next()
+      }
+      function end (next) {
+        res.end()
+        next()
+      }
+    }
+  },
+  admin: {
+    help: () => 'add or remove an admin for a channel or cabal-wide',
+    call: (cabal, res, arg) => {
+      if (!arg) {
+        res.info('usage: /admin (add|remove) (CHANNEL|@) KEY {REASON...}')
+        res.info('usage: /admin list (CHANNEL|@)')
```

Comment:
can we test out adding a mod/admin cabal-wide as the only option, to begin with? 

it feels like there's a lot going on already, so having granular mods might be too much to keep track of. (i'm fine with that granularity in cabal-core)

it seems to me that the default is normally going to be cabal-wide